pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -307,6 +306,30 @@ private void reloadSegment(String tableNameWithType, SegmentMetadata segmentMeta
     }
   }
 
+  /**
+   * Try to reload a mutable segment.
+   * @return true if the segment is mutable; false if the segment is immutable.
+   */
+  @VisibleForTesting
+  boolean reloadMutableSegment(String tableNameWithType, String segmentName,
```

Comment:
Maybe rename to reloadIfMutabelSegment() ?

Reply:
still prefer reloadMutableSegment as you suggested early on.